wp-config-docker.php

```diff
@@ -56,7 +56,7 @@ function getenv_docker($env, $default) {
 define( 'DB_HOST', getenv_docker('WORDPRESS_DB_HOST', 'mysql') );
 
 /** Database Charset to use in creating database tables. */
-define( 'DB_CHARSET', getenv_docker('WORDPRESS_DB_CHARSET', 'utf8') );
+define( 'DB_CHARSET', getenv_docker('WORDPRESS_DB_CHARSET', 'utf8mb4') );

```

Comment:
Shouldn't this default be changed in https://github.com/WordPress/WordPress/blob/6ad5c0c55373f898ef689dd4d4a54f635b03716a/wp-config-sample.php#L34-L35 too? (That's where this default value came from :confused:)

I would think that users who would be broken by such a change should already have a suitably-configured `wp-config.php` file.

Reply:
It's a bit convoluted to read, but I _think_ https://github.com/WordPress/WordPress/blob/f9cc35ebad82753e9c86de322ea5c76a9001c7e2/wp-admin/setup-config.php#L376-L380 could also be a reasonable justification for changing our default?

Reply:
You're right, I think it would be good to change it in the wp-config-sample mentioned above as well.
One reason I can think of why they might have kept it, could be interoperability with other databases. I usually use Postgres and the charset there is named just 'utf8'. However, that's just guessing.

I stumbled over this when migrating a Wordpress instance from a hosting provider to a docker-compose based setup. This other setup had hardcoded values which I changed to getenv. It had utf8mb4 set there and I was surprised that this was not the default, so i decided to create a PR here. I was lucky to find that here.

If I were not migrating but starting fresh, I would have had a look at wp-config but only use ENV vars, thus I would run into troubles, I'm afraid.

So, what can I do to resolve it this? Should I create a PR for the first file you linked as well?

Reply:
I think https://github.com/WordPress/WordPress/blob/f0078d043e0f2805c2400bd5e869eb3533712eec/wp-includes/wp-db.php#L724-L769 is probably relevant -- it looks like the WordPress internals will automatically opportunistically update `utf8` to `utf8mb4` based on the database's support level. :smile:

(Which then makes their default make more sense.)

Reply:
Interesting. And right below, they're also downgrading if it's set but not supported.

So, it wouldn't change anything in our case as long as only WordPress is concerned.

Feel free to merge or reject this PR however it pleases you. 😊